autoware_reference_system/test/generate_callback_traces.py

```diff
@@ -29,52 +29,55 @@
 
 # this file has @variables@ that are meant to be automatically replaced
 # by values using the `configure_file` CMake function during the build
+RUNTIME = int('@RUNTIME@')
 
 
 def generate_test_description():
-    env = os.environ.copy()
-    env['RCUTILS_CONSOLE_OUTPUT_FORMAT'] = '[{severity}] [{name}]: {message}'
-
     # replaced with cmake `configure_file` function
     rmw_impl = '@RMW_IMPLEMENTATION@'
     test_exe = '@TEST_EXECUTABLE@'
     test_exe_name = '@TEST_EXECUTABLE_NAME@'
-    timeout = int('@TIMEOUT@')
-    extra_timeout = timeout + 5
-
-    # specify rmw to use
-    env['RCL_ASSERT_RMW_ID_MATCHES'] = rmw_impl
-    env['RMW_IMPLEMENTATION'] = rmw_impl
 
     launch_description = LaunchDescription()
-    proc_under_test = ExecuteProcess(
-        cmd=[test_exe],
-        name=test_exe_name,
-        sigterm_timeout=str(extra_timeout),
-        output='screen',
-        env=env,
+
+    # see https://github.com/ros2/launch/issues/417
+    # have to use `SetEnvironmentVariable` to set env vars in launch description
+    envvar_rcutils_action = SetEnvironmentVariable(
+        'RCUTILS_CONSOLE_OUTPUT_FORMAT', '[{severity}] [{name}]: {message}')
+    envvar_rmw_action = SetEnvironmentVariable(
+        'RMW_IMPLEMENTATION', rmw_impl)
+    envvar_rclassert_rmw_action = SetEnvironmentVariable(
+        'RCL_ASSERT_RMW_ID_MATCHES', rmw_impl)
+
+    node_under_test = Node(
+        package='autoware_reference_system',
+        executable=test_exe,
+        output='screen'
     )
 
     trace_action = Trace(
-        session_name='profile_' + test_exe_name + '_' + str(timeout) + 's',
+        session_name='profile_callback_' + test_exe_name + '_' + str(RUNTIME) + 's',
         events_ust=[
             'lttng_ust_cyg_profile_fast:func_entry',
             'lttng_ust_cyg_profile_fast:func_exit',
             'lttng_ust_statedump:start',
             'lttng_ust_statedump:end',
             'lttng_ust_statedump:bin_info',
-            'lttng_ust_statedump:build_id',
+            'lttng_ust_statedump:build_id'
         ] + DEFAULT_EVENTS_ROS,
         events_kernel=[
-            'sched_switch',
+            'sched_switch'
         ],
         context_names=[
-            'ip',
+            'ip'
         ] + DEFAULT_CONTEXT,
```

Comment:
now with this MR https://gitlab.com/ros-tracing/ros2_tracing/-/merge_requests/263 you can do:

```python
context_names={
    'kernel': DEFAULT_CONTEXT,
    'userspace': DEFAULT_CONTEXT + ['ip'],
},
```

Reply:
we can add this back into this package once this is fully supported, maybe ill make an issue so we can track this.